### PR TITLE
Functions tutorial assemblyscript

### PR DESCRIPTION
**Pull request by rdraward**
This is a full copy of a Shopify CLI app with a Functions discounts tutorial example written in AssemblyScript.

- `api.ts` could use a bit of cleanup, but what we have works for the tutorial, I will do a quick pass and remove anything that is not used

### CODE REVIEW

File: packages/discounts-assemblyscript/extensions/volume/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Shopify Function development with Rust
```
Comment (airhorns) [inline packages/discounts-assemblyscript/extensions/volume/README.md:1]:
```suggestion
# Shopify Function development with AssemblyScript
```

File: packages/discounts-assemblyscript/extensions/volume/assembly/index.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    new DiscountValue(new Percentage(config.percentage)),
+    targets,
+    "Testing Shopify Functions using AssemblyScript"
+  );
+  return new FunctionResult(DiscountApplicationStrategy.FIRST(), discounts);
+}
+
+const input = Console.readAll()!;
+const output = main(input);
+Console.log(output);
```
Comment (airhorns) [inline packages/discounts-assemblyscript/extensions/volume/assembly/index.ts:61]: So little memory management! Wow! 🤘

File: packages/discounts-assemblyscript/extensions/volume/assembly/index.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
+// The entry file of your WebAssembly module.
```
Comment (airhorns) [inline packages/discounts-assemblyscript/extensions/volume/assembly/index.ts:1]: 🔥 ?

File: packages/discounts-assemblyscript/extensions/volume/schema.graphql [hunk 4]
```diff
@@ -0,0 +1,3587 @@
+schema {
```
Comment (airhorns) [inline packages/discounts-assemblyscript/extensions/volume/schema.graphql:1]: Is this file necessary?

**Review by airhorns — APPROVED** [review]
LGTM lets get it out!

File: packages/discounts-assemblyscript/extensions/volume/schema.graphql (hunk 4 above)
Comment (rdraward) [reply packages/discounts-assemblyscript/extensions/volume/schema.graphql:1]: It's a default file that comes with a new extension - I'm happy to remove it
